### ISSUE TITLE
Cross filters on bar chart only work when using dimensions

### ISSUE THREAD

**Issue by WibowoS26** [apache/superset#25334]
#### How to reproduce the bug

on superset 3.0.0

dashboard below before using cross filters  

![1](https://github.com/apache/superset/assets/143679513/50ede2c1-1942-4aa3-9920-8bb79c48bd2c)

dashboard below after using cross filters on blue bar chart

![2](https://github.com/apache/superset/assets/143679513/b062a320-84c6-4080-9910-61baec39f6f4)

same dashboard but using cross filters on the pie chart

![4](https://github.com/apache/superset/assets/143679513/50bcd580-28ca-4994-ae3f-8c934aef324e)

same dashboard but using dimensions on bar chart

![7](https://github.com/apache/superset/assets/143679513/66769cad-5f47-4779-b237-0b124eac2331)

### Expected results

cross filters work with bar chart without the need of dimensions

### Actual results

cross filters on bar chart only work when using dimensions

### Environment

(please complete the following information):

- browser type and version:
- superset version: `superset version`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by sfirke** [comment]
Interesting, people have reported cross filters not working for bar chart but I hadn't seen anyone point out it was dimension-specific.  Thanks for the bug report.

**Comment by WibowoS26** [comment]
@sfirke yes, because you can't reuse same column for metrics and dimensions on bar chart  you need to add another column to be used in dimensions

**Comment by mtrentz** [comment]
Running into the same problem (3.0) and also noticed it's dimension specific.

It seems that it tries to filter by the metric of the bar chart. When I click in one of the bars and hover over the filter applied in other charts it shows like this:

![image](https://github.com/apache/superset/assets/59535526/0856e59f-4561-4014-aee4-c42b7745bcae)

In contrast, doing it instead with a pie chart it shows on hover that is being filter by the correct dimension (name of salesperson in this case)

![Screenshot 2023-11-16 at 08 51 28](https://github.com/apache/superset/assets/59535526/9188562b-251f-4c65-b1ad-a1ea052e390e)

**Comment by sfirke** [comment]
Someone gave a good description of the bug in Slack, with videos.  https://apache-superset.slack.com/archives/C0170U650CQ/p1698853673073119?thread_ts=1684878085.826729&cid=C0170U650CQ

**Comment by 704998200** [comment]
I have also seen this issue.Is there any update?Or any solution?

**Comment by 704998200** [comment]
It seems to be related to DIMENSIONS.If category axis is not a date,you can set like this.It works for me.
![image](https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486)
![image](https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)

**Comment by sfirke** [comment]
I think the ideal fix would be: bar charts without dimension emit cross-filters as if the series was a dimension.  That seems to be the behavior people expect.  I have no idea how hard that is to implement.

**Comment by visharavana** [comment]
I've noticed this problem with 4.0 as well. Has anything changed recently? or a possible fix?
I want only one color, however if I use dimensions, the bar color will show every bar as every color.

**Comment by sfirke** [comment]
@visharavana this issue persists through 4.0.0. I do not know of a fix or workaround at this time.

**Comment by felipegranado** [comment]
Hi..

This issue persits on 4.0.1.

**Comment by sfirke** [comment]
Correct, it persists on the latest version 4.0.2 as well. I expect that if/when this is addressed, the pull request will link back here and close the issue.

**Comment by felipegranado** [comment]
Hi!

I didn't understand the X-Axis in this graph, this field is mandatory and at the same time it has no Cross-Filter action...

The funniest thing is that the legacy chart does not have this X-Axis and only the Dimensions field (and it is not mandatory).

Even because the X-Axis can become Y-Axis just by changing the format of the graph... so I didn't quite understand the meaning of this field. Dimensions field has everything we need.

I believe that making Dimensions mandatory instead of X-Axis will solve everything!

Thank you in advance for your efforts in resolving the issue. I'm just getting to know the tool now and it's very promising!

**Comment by smrodin** [comment]
> If category axis is not a date,you can set like this.It works for me.
> 
> (https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486) (https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)

Thanks, but when doing this, char bars get too narrow

**Comment by felipegranado** [comment]
> > If category axis is not a date,you can set like this.It works for me.
> > (https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486) (https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)
> 
> Thanks, but when doing this, char bars get too narrow

Oh.. I forget...

In CUSTOMIZE, you need choose "STACK" option...

**Comment by smrodin** [comment]
> > > If category axis is not a date,you can set like this.It works for me.
> > > (https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486) (https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)
> > 
> > 
> > Thanks, but when doing this, char bars get too narrow
> 
> Oh.. I forget...
> 
> In CUSTOMIZE, you need choose "STACK" option...

It works like a champ, thanks for your support

**Comment by heiwanokami** [comment]
I would like this to work so that the cross filter field is the category on x axis not the value (i have categrized data by warehouse zone stock and when i click the zone i expect to filter the neighbour table to be crossfiltered only to the item stored in that zone... curently the cross filter sends the number of pallets in the zone not the zone category... I think this may be common problem with the one reported

**Comment by luibrain** [comment]
I have the same issue. Cross-Filter only works with dimensions. There are situations where we do not need the date on x-axis.

**Comment by rusackas** [comment]
Anyone know if [#27941](https://github.com/apache/superset/pull/27941) had any impact here?

**Comment by 125m125** [comment]
> Anyone know if [#27941](https://github.com/apache/superset/pull/27941) had any impact here?

I just tested it with the 5.0.0rc3 and the behavior has changed a bit, but isn't really fully resolved. It now doesn't allow you to set a cross-filter anymore and says that it can't find a dimension:

![Image](https://github.com/user-attachments/assets/7df187ab-702c-413d-ae21-6b2af71006b8)

If you "force categorical" it allows you to drill down by the x value, but you still can't set a cross-filter:

![Image](https://github.com/user-attachments/assets/dd58d179-c710-42b8-8403-34b40cf81e0a)

So in both cases, while it isn't setting a "wrong"/"invalid" cross-filter, it still doesn't allow you to cross-filter by the x value.

**Comment by neeagl** [comment]
> If category axis is not a date,you can set like this.It works for me.
> 
> (https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486) (https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)

Thanks, this worked for me. Though I need to use 2 metrics for each x-axis item but when I select "Stacked", it puts them on top of each other. Any workaround? If I don't then it shows them as narrow bars

Another issue I noted while using multiple metrics is that Superset passes incorrect value to the filter (instead of region, it passes the name of the metric).

**Comment by carrarook** [comment]
Any updates on this?

**Comment by prashant-ux23** [comment]
Any update on this? We are also facing the same issue and not having the corss-filter in the bar chart makes it difficult to have an interactive dashboard in most cases.

**Comment by mousiyah** [comment]
Hello, the issue is still not fixed. Not possible to cross filter on bar chart. Please resolve this. There is no workaround on this

**Comment by stinehn** [comment]
Any update on this????

**Comment by erlitx** [comment]
> > If category axis is not a date,you can set like this.It works for me.
> > (https://github.com/apache/superset/assets/49085215/7f1cd5c8-2267-45fe-92f3-927ac9898486) (https://github.com/apache/superset/assets/49085215/b404037f-c14f-46e4-a2e5-83343be0c4b5)
> 
> Thanks, this worked for me. Though I need to use 2 metrics for each x-axis item but when I select "Stacked", it puts them on top of each other. Any workaround? If I don't then it shows them as narrow bars
> 
> Another issue I noted while using multiple metrics is that Superset passes incorrect value to the filter (instead of region, it passes the name of the metric).

In that case when bars are Staked - it start cross-filter by both metric itself and dimention - as a result crossfiltering on other charts give zero result

**Comment by rusackas** [comment]
Just let Claude take a crack at this issue... feel free to review/test!

**Comment by sfirke** [comment]
Wow @rusackas ! Nicely done. This was bothering a lot of people. I'll test it whenever the next release candidate for a major or minor version comes out

**Comment by syafiqmuda** [comment]
if I don't set a dimension here, I cant click on the bar to do cross filter. so how to solve this issue?

<img width="525" height="539" alt="Image" src="https://github.com/user-attachments/assets/d967865a-a0b6-456c-a777-b3754a3efa52" />

**Comment by sfirke** [comment]
@syafiqmuda this was fixed in the pull request that @rusackas linked above, should be in 6.1 or 7.0 whichever comes first

**Comment by syafiqmuda** [comment]
> [@syafiqmuda](https://github.com/syafiqmuda) this was fixed in the pull request that [@rusackas](https://github.com/rusackas) linked above, should be in 6.1 or 7.0 whichever comes first

I am on 6.0.0, latest already @sfirke 

<img width="281" height="813" alt="Image" src="https://github.com/user-attachments/assets/bdde4228-11a5-4635-903f-2989f624b6dd" />

**Comment by CamiloCarvajalPensemos** [comment]
> [@syafiqmuda](https://github.com/syafiqmuda) this was fixed in the pull request that [@rusackas](https://github.com/rusackas) linked above, should be in 6.1 or 7.0 whichever comes first

I'm currently on 6.1.0 and the problem persists. It's still necessary to set the Dimension with another name and choose "STACK" option.

**Comment by sfirke** [comment]
Thanks @CamiloCarvajalPensemos for the update, I see this PR open about it: https://github.com/apache/superset/issues/41102 so I'll reopen this.
